src/matlab2tikz.m

```diff
@@ -892,26 +892,52 @@ case guitypes()
 end
 % ==============================================================================
 function m2t = drawGridOfAxes(m2t, handle)
-    % draws the grids of an axes
-    %TODO: has{XYZ}Grid is always false without a good reason
-    hasXGrid = false;
-    hasYGrid = false;
-    hasZGrid = false;
-    if hasXGrid || hasYGrid || hasZGrid
-        matlabGridLineStyle = get(handle, 'GridLineStyle');
-        % Take over the grid line style in any case when in strict mode.
-        % If not, don't add anything in case of default line grid line style
-        % and effectively take Pgfplots' default.
-        defaultMatlabGridLineStyle = ':';
-        if m2t.cmdOpts.Results.strict ...
-                || ~strcmpi(matlabGridLineStyle,defaultMatlabGridLineStyle)
-            gls = translateLineStyle(matlabGridLineStyle);
+    % Draws the grids of an axis
+    % Check for major grids
+    hasGrid =  strcmpi(get(handle, 'XGrid'), 'on') ...
+            || strcmpi(get(handle, 'YGrid'), 'on') ...
+            || strcmpi(get(handle, 'ZGrid'), 'on');
+
+    % Check for minor grids
+    hasMinorGrid =  strcmpi(get(handle, 'XMinorGrid'), 'on') ...
+                 || strcmpi(get(handle, 'YMinorGrid'), 'on') ...
+                 || strcmpi(get(handle, 'ZMinorGrid'), 'on');
+
+    % Default matlab grid style ommited for brevity
+    % defaultGridLineStyle = ':';
```

Comment:
I would leave `defaultGridLineStyle` as variable. You can define it a bit further and directly do the compare:
```matlab
defaultGridLineStyle = ':';
hasLineStyle = strcmpi(gridLineStyle, defaultGridLineStyle);
```